pkg/db/utilities/utilities_test.go

```diff
@@ -0,0 +1,153 @@
+package utilities_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/transcom/mymove/pkg/db/utilities"
+	"github.com/transcom/mymove/pkg/models"
+	"github.com/transcom/mymove/pkg/services/mocks"
+	"github.com/transcom/mymove/pkg/testdatagen"
+	"github.com/transcom/mymove/pkg/testingsuite"
+	"github.com/transcom/mymove/pkg/unit"
+)
+
+type UtilitiesSuite struct {
+	testingsuite.PopTestSuite
+}
+
+func (suite *UtilitiesSuite) SetupTest() {
+	suite.DB().TruncateAll()
+}
+
+func TestUtilitiesSuite(t *testing.T) {
+	hs := &UtilitiesSuite{
+		PopTestSuite: testingsuite.NewPopTestSuite(testingsuite.CurrentPackage()),
+	}
+	suite.Run(t, hs)
+}
+
+func (suite *UtilitiesSuite) TestSoftDestroy_NotModel() {
+	accessCodeFetcher := &mocks.AccessCodeFetcher{}
+
+	err := utilities.SoftDestroy(suite.DB(), &accessCodeFetcher)
+
+	suite.Error(err)
+}
+
+func (suite *UtilitiesSuite) TestSoftDestroy_ModelWithoutDeletedAtWithoutAssociations() {
+	//model without deleted_at with no associations
+	user := testdatagen.MakeDefaultUser(suite.DB())
+
+	err := utilities.SoftDestroy(suite.DB(), &user)
+
+	suite.Error(err)
```

Comment:
Can we also add a check for the error message string?

```suggestion
	suite.Equal("Error message to validate", errors.Cause(err).Error())
```